src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/Common/DbConnectionStringCommon.cs

```diff
@@ -214,6 +214,136 @@ internal static string ColumnEncryptionSettingToString(SqlConnectionColumnEncryp
+        {
+            bool isSuccess = false;
+
+            if (StringComparer.InvariantCultureIgnoreCase.Equals(value, AttestationProtocolHGS))
+            {
+                result = SqlConnectionAttestationProtocol.HGS;
+                isSuccess = true;
+            }
+            else if (StringComparer.InvariantCultureIgnoreCase.Equals(value, AttestationProtocolAAS))
+            {
+                result = SqlConnectionAttestationProtocol.AAS;
+                isSuccess = true;
+            }
+            else
+            {
+                result = DbConnectionStringDefaults.AttestationProtocol;
+            }
+
+            return isSuccess;
+        }
+
+        internal static bool IsValidAttestationProtocol(SqlConnectionAttestationProtocol value)
+        {
+            Debug.Assert(Enum.GetNames(typeof(SqlConnectionAttestationProtocol)).Length == 3, "SqlConnectionAttestationProtocol enum has changed, update needed");
+            return value == SqlConnectionAttestationProtocol.NotSpecified 
+                || value == SqlConnectionAttestationProtocol.HGS 
+                || value == SqlConnectionAttestationProtocol.AAS;
+
+        }
+
+        internal static string AttestationProtocolToString(SqlConnectionAttestationProtocol value)
+        {
+            Debug.Assert(IsValidAttestationProtocol(value), "value is not a valid attestation protocol");
+
+            switch (value)
+            {
+                case SqlConnectionAttestationProtocol.HGS:
+                    return AttestationProtocolHGS;
+                case SqlConnectionAttestationProtocol.AAS:
+                    return AttestationProtocolAAS;
+                default:
+                    return null;
+            }
+        }
+
+        internal static SqlConnectionAttestationProtocol ConvertToAttestationProtocol(string keyword, object value)
+        {
+            if (null == value)
+            {
+                return DbConnectionStringDefaults.AttestationProtocol;
+            }
+
+            string sValue = (value as string);
+            SqlConnectionAttestationProtocol result;
+
+            if (null != sValue)
+            {
+                if (TryConvertToAttestationProtocol(sValue, out result))
+                {
+                    return result;
+                }
+
+                // try again after remove leading & trailing whitespaces.
+                sValue = sValue.Trim();
+                if (TryConvertToAttestationProtocol(sValue, out result))
+                {
+                    return result;
+                }
+
+                // string values must be valid
+                throw ADP.InvalidConnectionOptionValue(keyword);
+            }
+            else
+            {
+                // the value is not string, try other options
+                SqlConnectionAttestationProtocol eValue;
+
+                if (value is SqlConnectionAttestationProtocol)
```

Comment:
This is more of a question. When you do `value is SqlConnectionAttestationProtocol` doesn't that already mean it must be a `SqlConnectionAttestationProtocol` enum and it must be 0, 1, or 3? Is the check in line 335 redundant in this case?

Reply:
I can see this logic being used for other enum type connection properties such as ConvertToAuthenticationType and ConvertToColumnEncryptionSetting. Not sure why they are designed in this way though.